freki.go

```diff
@@ -512,10 +516,10 @@ func (p *Processor) onPacket(rawPacket *netfilter.RawPacket) (err error) {
 				}
 
 				// FYI: when i don't respond to a SYN, then a duplicate SYN is sent
-				ck := NewConnKeyByEndpoints(srcIP, srcPort)
 				p.Connections.Register(ck, rule, srcIP.String(), srcPort.String(), tcp.DstPort)
 			}
 
+			p.Connections.updatePacketTime(ck)
```

Comment:
I think you can move this into an `else` from the above `if` since `p.Connections.Register` sets the time.